Mostekcm add apps and client grants

mostekcm opened the pull request:
This is the pull request to add resource servers and client grants.

The client grants are done as part of the meta data for the client.  This greatly reduces the areas for errors as it enforces the existence of a client and only allows audience/scope pairs.

Verlic commented:
MD5: c4103f122d27677c9db144cae1394a66

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **2**
Affected Modules: **uglify-js, uglify-js**
Summary:

- Regular Expression Denial of Service
- Incorrect Handling of Non-Boolean Comparisons During Minification


## Detailed report
#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-source-control-extension-tools@2.2.0 > auth0-oauth2-express@1.2.1 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-source-control-extension-tools@2.2.0 > auth0-oauth2-express@1.2.1 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

mostekcm commented:
The large amount of changes was due to the fact that I refactored clients.  I moved the common code between clients and resource servers to a "configurables" module.  This made it quite simple to add resource servers.
